Makefile

```diff
@@ -10,10 +10,24 @@
 # or export those into environment, or pass on the command line e.g.
 # make all DEBUG=1 CXX=g++-12
 
+HOST_OS = $(shell uname -s)
+
+ifeq ($(HOST_OS), Darwin)
+CXX = g++-14 # from homebrew, needed for 128-bit floats
+else
+CXX = g++
+endif
+
 COMMON_FLAGS = -Wall -std=c++20
 # -static-libstdc++ -static-libgcc
 # -fext-numeric-literals
 
+ifeq ($(HOST_OS), Darwin)
+OPENCL_LIBS = -framework OpenCL
```

Comment:
That's a weird way to link to a library, but it's needed and GCC supports it. Copied form mfakto.